### DISCUSSION TITLE
How to make alevin-fry --expect cells to be better at guessing the number of cells?

### DISCUSSION THREAD

**Discussion by ljudevitluka** [COMBINE-lab/alevin-fry#101] [Q&A]
Hi all,
I am working on the scRNAseq dataset generated with the 10x Genomics platform. We aimed to collect 10, 000 cells per experiment, and we mapped the reads to the transcriptome (since no genomic reference is available at the moment). Everything seems to be working more or less ok with alevin-fry, but in my samples, I always get the overestimation of the cell number (as per the graph bellow). I am aware we could use --force-cells, but would be more comfortable if the knee is detected by the --expect-cells flag. I tried calling --expect-cells 5000, 8000, and 10000 cells but the results were always overestimated. 
My question is: Is there a way to make --expect-cells more sensitive? If I use --force-cells how will this affect my downstream analysis if some of the samples are carrying noise? If there is no automated way, is it better to take a bit of noise and hope it will get filtered out based on the further cell QC (mitoDNA genes content, ribosomal content etc.) or to be on the secure side and take a bit less cell with a risk of loss of the cells that have a lower number of genes expressed? 
Thank you very much for your help and opinions :) Hopefully, this question is not out of the scope of discussion. 
All the best, Luka
![image](https://user-images.githubusercontent.com/55898920/215842266-21e2c1b2-bbeb-4033-84d6-5c02f10018d9.png)

**Comment by georginaalbadri** [comment]
Hi Luka, 
I appreciate this was quite a while ago, but did you find a solution? I am facing something similar - using the --unfiltered list option is only giving me 10% of the cells that cellranger does (1k vs 10k), whereas using the --expect-cells handle is greatly overestimating (100k+). I need something in the middle! I know there are another couple of options, which I will try, but it seems like a common problem and I can't find any documentation that discusses this in depth. 
Thanks,
Georgina

**Reply by rob-p** [reply]
Hi @georginaalbadri,

  Can you share some of the output for when you provide the `--unfiltered` option?  Generally, the only thing left out of the unfiltered list are cells with < 10 *reads* (i.e. cells from which you are certainly unlikely to get any useful information).  This might be indicative of CellRanger massively overestimating the number of valid cells, or, perhaps, something more mundane like accidentally specifying the wrong chemistry.

--Rob

**Reply by georginaalbadri** [reply]
Hi Rob,

Thanks a lot for your reply, you have triggered me to look into the chemistry further and I've discovered that 10x uses a different list of barcodes for their multiome sequencing, which is where my data is from. I will try to download this list and input it manually, which I think I saw an option for in the simpleaf wrapper I'm using. 

Regarding output, I don't have any alevin.log files in my output folder, am I looking for something else to see stats like % cells recovered? 

Thanks,
Georgina

**Reply by rob-p** [reply]
Hi @georginaalbadri,

  Indeed, the barcode list can make a big difference. In the meantime, what you'd be looking for are `.json` files in the output directory.  They are typically named as the stage in the pipeline; so things like `generate_permit_list.json`, `collate.json`, `quant.json` and `meta_info.json`.  These contain the specific commands run and some summary output about each.

Best,
Rob

**Reply by georginaalbadri** [reply]
Hi @rob-p,

It's certainly made a difference, from 1k to 100k cells with unfilered-pl and the correct multiome barcode list. Is that many expected, and then quality filters will get it closer to the 10k we're expecting? I'm still worried I'm doing something wrong! 

Thanks,
Georgina

**Reply by rob-p** [reply]
Hi @georginaalbadri,

  Indeed; this *is* expected.  When one uses the `--unfiltered` option, this quantifies *all* cells in the sample matching a known barcode with >= 10 reads.  Many of these cells will, in reality, be of low quality or even empty droplets, and will be filtered out by the normal filtering procedure.  Having 1-2 orders of magnitude more unfiltered barcodes than filtered is completely normal for this kind of data!

Best,
Rob
